contracts/IColonyNetwork.sol

```diff
@@ -189,4 +189,6 @@ contract IColonyNetwork {
   /// @notice Register a "colony.joincolony.eth" label. Can only be called by a Colony.
   /// @param subnode The keccak256 hash of the label to register.
   function registerColonyLabel(bytes32 subnode) public;
+
+  function setMiningResolver(address miningResolverAddress) public; 
```

Comment:
Given the contracts are relatively short lived, if we wanted to upgrade the mining cycle contracts while not in recovery mode, I would in the first instance try to just change `miningCycleResolver` on colonyNetwork and have all future instances of the contract be the upgraded ones, rather than upgrading the active ones in-place.

If we do need to upgrade them in place though, then yeah, I'm on board with only being able to do that in recovery mode.